src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -45,15 +152,27 @@ public DynamicJsonValue ToJson()
             {
                 [nameof(Members)] = new DynamicJsonArray(Members),
                 [nameof(Promotables)] = new DynamicJsonArray(Promotables),
-                [nameof(Watchers)] = new DynamicJsonArray(Watchers)
+                [nameof(Watchers)] = new DynamicJsonArray(Watchers.Select(w => w.ToJson())),
+                [nameof(NameToUrlMap)] = DynamicJsonValue.Convert(NameToUrlMap)
             };
         }
 
         public void RemoveFromTopology(string delDbFromNode)
         {
             Members.Remove(delDbFromNode);
             Promotables.Remove(delDbFromNode);
-            Watchers.Remove(delDbFromNode);
+        }
+
+        public bool IsItMyTask(IDatabaseTask task, string nodeTag)
+        {
+            var myPosition = Members.FindIndex(s => s == nodeTag);
+            if (myPosition == -1) //not a member
+            {
+                return false;
+            }
+
+            //TODO : ask Oren here about suspentions.. (review comments in github)
```

Comment:
This should be part of the heartbeat work, which should be another list noticed as dead.